packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -147,7 +164,16 @@ export class SerializedStateManager {
 	private readonly mc: MonitoringContext;
 	private snapshot: ISnapshotInfo | undefined;
 	private latestSnapshot: ISnapshotInfo | undefined;
-	private _refreshSnapshotP: Promise<number> | undefined;
+	private readonly refreshTracker = new RefreshPromiseTracker(
+		// eslint-disable-next-line unicorn/consistent-function-scoping
+		(error) =>
+			this.mc.logger.sendErrorEvent(
+				{
+					eventName: "RefreshLatestSnapshotFailed",
+				},
+				error,
+			),
+	);
```

Comment:
Not a huge fan of the RefreshPromiseTracker - `Promise | undefined` seems plainer to the reader.  Also keeps error handling adjacent to promise creation.

Reply:
i'd rather keep it. it prevents errors with how the member is managed and ensure only one promise is tracked at a time. with the variable it could just be overwritten without notice

Reply:
If your concern is overwriting, a `set()` with similar check is the better solution

Reply:
it is better to minimize mutable state within the larger class, and isolate and share error handling which is what the RefreshPromiseTracker does